Source/Core/HeadingPitchRoll.js

```diff
@@ -0,0 +1,145 @@
+/*global define*/
+define([
+    './defaultValue',
+    './defined',
+    './DeveloperError',
+    './freezeObject',
+    './Math',
+    './defineProperties'
+], function (
+        defaultValue,
+        defined,
+        DeveloperError,
+        freezeObject,
+        CesiumMath,
+        defineProperties) {
+    "use strict";
+
+    /**
+     * heading pitch roll type. Heading is the rotation about the
+     * negative z axis. Pitch is the rotation about the negative y axis. Roll is the rotation about
+     * the positive x axis.
+     * @alias HeadingPitchRoll
+     * @constructor
+     *
+     * @param {Number} [heading=0.0] The heading component in radians.
+     * @param {Number} [pitch=0.0] The pitch component in radians.
+     * @param {Number} [roll=0.0] The roll component in radians.
+     */
+    var HeadingPitchRoll = function (heading, pitch, roll) {
+        this._heading = 0.0;
+        this._pitch = 0.0;
+        this._roll = 0.0;
+
+        this.heading = defaultValue(heading, 0.0);
+        this.pitch = defaultValue(pitch, 0.0);
+        this.roll = defaultValue(roll, 0.0);
+    };
+
+    defineProperties(HeadingPitchRoll.prototype, {
+        /**
+         * The heading component in radians.
+         * @type {Number}
+         * @default 0.0
+         */
+        heading: {
+            get: function () {
+                return this._heading;
+            },
+            set: function (value) {
+                value %= (2 * CesiumMath.PI);
+                this._heading = value;
+            }
+        },
+        /**
+         * The pitch component in radians. must be between -PI/2 and PI/2 strictly
+         * @type {Number}
+         * @default 0.0
+         */
+        pitch: {get: function () {
+                return this._pitch;
+            },
+            set: function (value) {
+                if ((value > -CesiumMath.PI_OVER_TWO) &&
+                        (value < CesiumMath.PI_OVER_TWO)) {
+                    this._pitch = value;
+                }
+            }
+        },
+        /**
+         * The roll component in radians.
+         * @type {Number}
+         * @default 0.0
+         */
+        roll: {
+            get: function () {
+                return this._roll;
+            },
+            set: function (value) {
+                value %= (2 * CesiumMath.PI);
+                this._roll = value;
+            }
+        }
+    });
+
+    /**
+     * Computes the heading, pitch and roll from a quaternion (see http://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles )
+     *
+     * @param {Quaternion} quaternion The quaternion to retrieve heading,pitch and roll in radians.
```

Comment:
Revise to "The quaternion from which to retrieve heading, pitch, and roll, all expressed in radians."